source/agora/common/Serializer.d

```diff
@@ -43,16 +43,20 @@ public mixin template DefaultSerializer ()
 ///
 unittest
 {
+    import agora.common.Amount;
     static struct Foo
     {
-        uint a;
-        ubyte[] b;
-
+        ubyte a;
+        ushort b;
+        uint c;
+        ulong d;
+        Amount e;
+        ubyte[] f;
         mixin DefaultSerializer!();
     }
-
-    const Foo f = Foo(uint.max, [1, 2, 3]);
-    assert(serializeFull(f) == [255, 255, 255, 255, 3, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3]);
+    const Foo f = Foo(1, ushort.max, uint.max, ulong.max, Amount(100), [1, 2, 3]);
+    assert(serializeFull(f) == [1, 253, 255, 255, 254, 255, 255, 255, 255]
+    ~ [255, 255, 255, 255, 255, 255, 255, 255, 255, 100, 3, 1, 2, 3]);
```

Comment:
Same comment as the deserialization example, please make it more readable

Reply:
Sorting Read-Easy Done.